### PR TITLE
Rating foundation

### PR DESCRIPTION
**Pull request by BernhardPosselt**
Adds the backend basics without any controller/view hooks. This will allow the current API to expose a default rating of 0.5 which can be adjusted later on.

Another PR will follow which will add the controllers and gui interaction. 

The basic idea is to calculate a new app rating after every new rating (because its very expensive) for both recent (in the last 90 days) and overall.

### CODE REVIEW

**Comment by BernhardPosselt** [comment]
@janibonnevier @LukasReschke @oparoz

File: nextcloudappstore/core/models.py [hunk 1]
```diff
@@ -196,6 +202,58 @@ def _latest(self, releases):
             return None
 
 
+class AppRating(TranslatableModel):
+    app = ForeignKey('App', related_name='app', verbose_name=_('App'),
```
Comment (janibonnevier) [inline nextcloudappstore/core/models.py]: I believe `related_name` should be something like `app_rating`. Same on line 209.

Comment (janibonnevier) [inline nextcloudappstore/core/models.py]: Or rather, "ratings".

File: nextcloudappstore/core/rating.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+
+def compute_rating(ratings: List[float], threshold: int = 10) -> float:
+    """
+    Turns a list of ratings into a score, 0 being the lowest and 1 being the
+    highest. The idea is that everyone can use his own custom scale, e.g.
+    a 5 star rating would multiply the value by 5. Also a too low number of
+    ratings will default to a constant value.
+    :param ratings: list of floats from 0 to 1
+    :param threshold: if the amount of ratings is lower than this number
```
Comment (janibonnevier) [inline nextcloudappstore/core/rating.py]: "lower than or equal to" according to the if-statement.

File: nextcloudappstore/core/models.py [hunk 3]
```diff
@@ -196,6 +202,58 @@ def _latest(self, releases):
...
+        super().save(*args, **kwargs)
+        # update rating on the app
+        app = self.app
+        day_range = settings.RATING_RECENT_DAY_RANGE
+        threshold = settings.RATING_THRESHOLD
+        app.rating_recent = self._compute_app_rating(app, day_range, threshold)
+        app.rating_overall = self._compute_app_rating(app, threshold=threshold)
+        app.save()
+
+    def _compute_app_rating(self, app: App, days: int = -1,
```
Comment (janibonnevier) [inline nextcloudappstore/core/models.py]: Isn't `app` always the same as `self.app`? In other words: how come this method is basically static? And if we want it to be static, does it belong in a `Model` class?

**Comment by janibonnevier** [comment]
Apart from my nitpick on IRC, :+1:

**Pull request merged**